pyblish_nuke/lib.py

```diff
@@ -215,3 +230,54 @@ def _show_no_gui():
 
 def where(program):
     """DEPRECATED"""
+
+
+def _nuke_set_zero_margins(widget_object):
+    """Remove Nuke margins when docked UI
+    .. _More info:
+        https://gist.github.com/maty974/4739917
+    """
+    parentApp = QtWidgets.QApplication.allWidgets()
+    parentWidgetList = []
+    for parent in parentApp:
+        for child in parent.children():
+            if widget_object.__class__.__name__ == child.__class__.__name__:
+                parentWidgetList.append(
+                    parent.parentWidget())
+                parentWidgetList.append(
+                    parent.parentWidget().parentWidget())
+                parentWidgetList.append(
+                    parent.parentWidget().parentWidget().parentWidget())
+
+                for sub in parentWidgetList:
+                        for tinychild in sub.children():
+                            try:
+                                tinychild.setContentsMargins(0, 0, 0, 0)
+                            except:
+                                pass
+
+
+class pyblish_nuke_dockwidget(QtWidgets.QWidget):
+    def __init__(self):
+        QtWidgets.QWidget.__init__(self)
+        QtWidgets.QVBoxLayout(self)
+        self.setObjectName("pyblish_nuke.dock")
+
+
+def dock_window(widget):
+
+    # delete existing dock
```

Comment:
Yes please, it does clean-up on close.

- https://github.com/pyblish/pyblish-lite/blob/master/pyblish_lite/window.py#L914